src/lazy/text/matched.rs

```diff
@@ -58,31 +60,84 @@ pub(crate) enum MatchedValue {
     Symbol(MatchedSymbol),
     Blob(MatchedBlob),
     Clob(MatchedClob),
-    List,
-    SExp,
-    Struct,
+    List(&'top [LazyRawValueExpr<'top, D>]),
+    SExp(&'top [LazyRawValueExpr<'top, D>]),
+    Struct(&'top [LazyRawFieldExpr<'top, D>]),
+}
+
+impl<'top, D: LazyDecoder> PartialEq for MatchedValue<'top, D> {
```

Comment:
🗺️ Storing slices of `LazyRawFieldExpr` meant that we could no longer derive an implementation of `PartialEq`.